front/src/components/UI/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  SidebarStyled,
+  SidebarList,
+  SidebarListItem,
+  SidebarListItemIcon,
+  SidebarListItemTitle,
+} from "./Sidebar.styled";
+
+import { MaintenanceAreaSidebarInterface } from "../../App/MaintenanceArea/MaintenanceArea.sidebarData";
+
+interface SidebarProps {
+  data: MaintenanceAreaSidebarInterface[];
+  onClick?: React.MouseEventHandler<HTMLInputElement>;
+}
+
+const Sidebar = (props: SidebarProps) => {
+  const myRouter = (link: string) => {
+    const mainRoutes = [
+      "/",
+      "help",
+      "maintenance",
+      "statistics",
+      "sales",
+      "product",
+    ];
+
+    let lastPath = window.location.pathname.split("/").pop();
+    let newUrl =
+      window.location.pathname.split("/")[
+        window.location.pathname.split("/").length - 2
+      ];
+
+    mainRoutes.find((item) => {
+      if (lastPath === item)
+        return (window.location.pathname = lastPath + link);
+      else if (newUrl === item)
+        return (window.location.pathname = newUrl + link);
+    });
+  };
+
+  return (
+    <SidebarStyled>
+      <SidebarList>
+        {props.data.map((val, key) => {
+          return (
+            <SidebarListItem key={key} onClick={() => myRouter(val.link)}>
```

Comment:
myRouter to rzeczownik a nazwy funkcji powinny być czasownikami bo opisuja jakąs akcje/czynność